registration/include/pcl/registration/impl/transformation_estimation_eig.hpp

```diff
@@ -0,0 +1,238 @@
+pcl::registration::TransformationEstimationEIG<PointSource, PointTarget, Scalar>::estimateRigidTransformation (
+  const pcl::PointCloud<PointSource> &cloud_src,
+  const std::vector<int> &indices_src,
+  const pcl::PointCloud<PointTarget> &cloud_tgt,
+  Matrix4 &transformation_matrix) const
+{
+  if ( indices_src.size () != cloud_tgt.points.size () )
+  {
+    PCL_ERROR ( "[pcl::TransformationEIG::estimateRigidTransformation] Number or points in source (%lu) differs than target (%lu)!\n", indices_src.size (), cloud_tgt.points.size () );
+    return;
+  }
+
+  ConstCloudIterator<PointSource> source_it ( cloud_src, indices_src );
+  ConstCloudIterator<PointTarget> target_it ( cloud_tgt );
+  estimateRigidTransformation ( source_it, target_it, transformation_matrix);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationEIG<PointSource, PointTarget, Scalar>::estimateRigidTransformation (
+  const pcl::PointCloud<PointSource> &cloud_src,
+  const std::vector<int> &indices_src,
+  const pcl::PointCloud<PointTarget> &cloud_tgt,
+  const std::vector<int> &indices_tgt,
+  Matrix4 &transformation_matrix) const
+{
+  if ( indices_src.size () != indices_tgt.size () )
+  {
+    PCL_ERROR ( "[pcl::TransformationEstimationEIG::estimateRigidTransformation] Number or points in source (%lu) differs than target (%lu)!\n", indices_src.size (), indices_tgt.size () );
+    return;
+  }
+
+  ConstCloudIterator<PointSource> source_it ( cloud_src, indices_src );
+  ConstCloudIterator<PointTarget> target_it ( cloud_tgt, indices_tgt );
+  estimateRigidTransformation ( source_it, target_it, transformation_matrix);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> void
+pcl::registration::TransformationEstimationEIG<PointSource, PointTarget, Scalar>::estimateRigidTransformation (
+  const pcl::PointCloud<PointSource> &cloud_src,
+  const pcl::PointCloud<PointTarget> &cloud_tgt,
+  const pcl::Correspondences &correspondences,
+  Matrix4 &transformation_matrix) const
+{
+  ConstCloudIterator<PointSource> source_it ( cloud_src, correspondences, true );
+  ConstCloudIterator<PointTarget> target_it ( cloud_tgt, correspondences, false );
+  estimateRigidTransformation ( source_it, target_it, transformation_matrix);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationEIG<PointSource, PointTarget, Scalar>::estimateRigidTransformation (
+  ConstCloudIterator<PointSource> &source_it,
+  ConstCloudIterator<PointTarget> &target_it,
+  Matrix4 &transformation_matrix) const
+{

```

Comment:
This function is quite long and seems to b made up of stages. It'd benefit from some refactoring or at least comments.